include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,603 @@
+ * @param unit_quaternion
+ * @return Eigen::Matrix3<Scalar>
+ */
+template<typename Scalar = float>
+inline Eigen::Matrix3<Scalar> QuaternionToRotationMatrix(
+    const Eigen::Quaternion<Scalar> &unit_quaternion)
+{
+  return QuaternionToRotationMatrix(unit_quaternion, Scalar(2));
+}
+
+/**
+ * @brief Constructs a Extrinsic X-Y-Z euler angle representation from a 
+ *        normalized quaternion and a rotation matrix
+ * @note See the cited Drake code for details
+ * @cite Russ Tedrake and the Drake Development Team, "Drake: Model-based design
+ *       and verification for robotics" (2019)
+ * @link https://github.com/RobotLocomotion/drake/blob/5022e3832e90f988f1b6d3015b2794ea26e527b1/math/roll_pitch_yaw.cc#L112 @endlink
+ *
+ * @tparam Scalar[optional] numeric type (default float)
+ * @param quaternion
+ * @param R
+ * @return EulerAngles<Scalar>
+ */
+template<typename Scalar = float>
+EulerAngles<Scalar> QuaternionAndRotationMatrixToEulerAngles(
+    const Eigen::Quaternion<Scalar> &quaternion,
+    const Eigen::Matrix3<Scalar> &R)
+{
+  using std::abs;
+  using std::atan2;
+  using std::sqrt;
+
+  // Calculate q2 using lots of information in the rotation matrix.
+  // Rsum = abs( cos(q2) ) is inherently non-negative.
+  // R20 = -sin(q2) may be negative, zero, or positive.
+  const Scalar R22 = R(2, 2);
+  const Scalar R21 = R(2, 1);
+  const Scalar R10 = R(1, 0);
+  const Scalar R00 = R(0, 0);
+  const Scalar Rsum = sqrt((R22 * R22 + R21 * R21 + R10 * R10 + R00 * R00) / 2);
+  const Scalar R20 = R(2, 0);
+  const Scalar q2 = atan2(-R20, Rsum);
+
+  // Calculate q1 and q3 from Steps 2-6 (see link).
+  const Scalar e0 = quaternion.w(), e1 = quaternion.x();
+  const Scalar e2 = quaternion.y(), e3 = quaternion.z();
+  const Scalar yA = e1 + e3, xA = e0 - e2;
+  const Scalar yB = e3 - e1, xB = e0 + e2;
+  const Scalar epsilon = Eigen::NumTraits<Scalar>::epsilon();
+  const auto isSingularA = abs(yA) <= epsilon && abs(xA) <= epsilon;
+  const auto isSingularB = abs(yB) <= epsilon && abs(xB) <= epsilon;
+  const Scalar zA = isSingularA ? Scalar{0.0} : atan2(yA, xA);
+  const Scalar zB = isSingularB ? Scalar{0.0} : atan2(yB, xB);
+  Scalar q1 = zA - zB; // First angle in rotation sequence.
+  Scalar q3 = zA + zB; // Third angle in rotation sequence.
+
+  // If necessary, modify angles q1 and/or q3 to be between -pi and pi.
+  q1 = q1 > M_PI ? q1 - 2 * M_PI : q1;
+  q1 = q1 < -M_PI ? q1 + 2 * M_PI : q1;
+  q3 = q3 > M_PI ? q3 - 2 * M_PI : q3;
+  q3 = q3 < -M_PI ? q3 + 2 * M_PI : q3;
+
+  // Return in conventional SpaceXYZ q1, q2, q3 (roll-pitch-yaw) order
+  // (which is equivalent to BodyZYX q3, q2, q1 order).
+  return EulerAngles<Scalar>(q1, q2, q3);
+}
+
+/**
+ * @brief Constructs euler angles from a normalized quaternion
+ *
+ * @tparam Scalar[optional] numeric type (default float)
+ * @param unit_quaternion
+ * @return EulerAngles<Scalar>
+ */
+template<typename Scalar = float>
+EulerAngles<Scalar> QuaternionToEulerAngles(
```

Comment:
```suggestion
EulerAngles<Scalar> QuaternionToDefaultEulerAngles(
```

Reply:
Resolved in d3d867197e2aa610a80dc502b3811b3b1d9302de